### PR TITLE
Default field values update

### PR DESCRIPTION
**Pull request by shramroy**

### CODE REVIEW

File: device/ethernet.yaml
Comment (ajbalogh) [inline device/ethernet.yaml:22]: is this the commonly accepted default?

File: flow/flow.yaml
Comment (ajbalogh) [inline flow/flow.yaml:28]: Disagree with this, the packet list is what is configured, there is also no notation in openapi to explicitly define a default item for the list.

File: layer1/layer1.yaml
Comment (ajbalogh) [inline layer1/layer1.yaml:42]: Disagree as it was changed to be 10 from 1.

File: layer1/layer1.yaml
Comment (ajbalogh) [inline layer1/layer1.yaml:60]: This doesn't appear to be the common default.

**Review by ajbalogh — CHANGES_REQUESTED** [review]

File: flow/flow.yaml
Comment (ashutshkumr) [reply flow/flow.yaml:28]: @ajbalogh @ankur-sheth won't it be more user-friendly if we configure a default ethernet header under the hood when no packet headers are provided ? not being able to specify default for lists is a limitation in openapi, hence we chose to mention it in description.

File: layer1/layer1.yaml
Comment (shramroy) [reply layer1/layer1.yaml]: Changed

File: layer1/layer1.yaml
Comment (shramroy) [reply layer1/layer1.yaml]: Changed

File: device/ethernet.yaml
Comment (shramroy) [reply device/ethernet.yaml]: changed

File: flow/flow.yaml
Comment (ajbalogh) [reply flow/flow.yaml]: Reviewed the openapi spec and you can specify a default in this case by setting the minimum array length to 1.  
Be aware that every packet header that gets added now defaults to ethernet instead of custom.  

```yaml
in flow.yaml add minItems to the packet property

        packet:
          description: >-
            The header is a list of traffic protocol headers.
            The order of traffic protocol headers assigned to the list is the
            order they will appear on the wire.
          type: array
          minItems: 1
          items:
            $ref: './packet-headers/header.yaml#/components/schemas/Flow.Header'

in header.yaml change the choice property default to ethernet

    Flow.Header:
      description: >-
        Container for all traffic packet headers
      type: object
      required: [choice]
      properties:
        choice:
          type: string
          enum: [custom, ethernet, vlan, vxlan, ipv4, ipv6, pfcpause, ethernetpause, tcp, udp, gre, gtpv1, gtpv2, arp, icmp, icmpv6, ppp, igmpv1]
          default: ethernet

```

Comment (ajbalogh) [reply flow/flow.yaml]: Everything else lgtm but I will hold off on my review until this conversation is resolved

**Review by ajbalogh — COMMENTED** [review]
LGTM except for default ethernet packet header discussion

File: flow/flow.yaml
Comment (shramroy) [reply flow/flow.yaml]: I have updated the changes according.

Comment (ashutshkumr) [reply flow/flow.yaml]: @shramroy could you please check and answer expected behavior here (I'm trying to understand how minItems works) ?
- user specifies flow without specifying packets
- user specifies flow with packets where list is empty

In both the cases I'm assuming we'll end up with packets containing list of one header, which defaults to ethernet (at least controller needs to ensure it).
If so, the change LGTM.

**Review by ajbalogh — APPROVED** [review]
lgtm - if you want the checks to pass modify lines 61-62 to add the-f flag in the do.sh to look like this:

```
    && ln -sf /usr/bin/python3 /usr/bin/python \
    && ln -sf /usr/bin/pip3 /usr/bin/pip \
```

File: flow/flow.yaml
Comment (shramroy) [reply flow/flow.yaml]: @ashutshkumr yes 3 scenarios covered.

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**
